File: Day48_RodCut.cpp
#include <bits/stdc++.h>
using namespace std;
ifstream in("in.txt");
ofstream out("out.txt");

int cutRod(int price[], int n)
{
    vector<int> dp(n + 1);
    for (int i = 1; i <= n; i++)
    {
        for (int j = i; j <= n; j++)
        {
            dp[j] = max(dp[j], dp[j - i] + price[i - 1]);
        }
    }
    return dp[n];
}

int main()
{

    return 0;
}

File: Day48_RodCut.py
def cutRod(price, n):
    dp = [0] * (n+1)
    for i in range(1, n+1):
        for j in range(i, n+1):
            dp[j] = max(dp[j], dp[j-i] + price[i-1])
    return dp[n]


p = [3, 5, 8, 9, 10, 17, 17, 20]
n = 8
print(cutRod(p, n))